src/op_utils.rs

```diff
@@ -170,6 +170,44 @@ fn test_i32_from_u8() {
     assert_eq!(i32_from_u8(&[0x7d, 0xcc, 0x55, 0x88, 0xf3]), None);
 }
 
+pub fn u64_from_bytes(buf: &[u8]) -> u64 {
```

Comment:
This https://doc.rust-lang.org/std/primitive.u64.html#method.from_be_bytes almost works, but you'd have to copy to an 0 array first. Might be faster, who knows. Probably doesn't matter.

Reply:
yeah, I considered using this function, I kind of dismissed it as I realized it took 8 bytes unconditionally. But it might end up being less code to copy it

Reply:
the logic to set up a temporary buffer for this is about as complex as this code. It might be slightly faster but I don't think there's really a strong enough motivation to make this change now. I think we should look into all of these functions (for `i32`, `u32` and `u64`) to use a temporary buffer and the built-in cast in the future.